source/elmer/HFReaderModule.f90

```diff
@@ -129,7 +130,16 @@ FUNCTION heatFluxOnNodes(Model, n, t) RESULT(hf)
         print *, "Found negative HF value..."
         hf = 0.0
     ENDIF
-
+    
+    ! Read the heat flux value (q_flow) from the boundary condition
+    q_flow = getConstReal(BC, 'q_flow', GotIt)
+    IF (.NOT. GotIt) THEN
+        CALL Fatal('heatFluxOnNodes', 'q_flow not found in boundary condition')
+    END IF    
+    
+    ! Add constant heat flux to boundary
+    hf = hf + q_flow
+    
```

Comment:
I would suggest you do not `Call Fatal()` if `getConstReal(BC, 'q_flow', GotIt)` returns a fail value.  This way people can use their old SIFs (that dont have q_flow) and it will just default to 0 if not found.  This makes the code backwards compatible with previous SIF versions.

So something like this:
```f90
    ! Read the heat flux value (q_flow) from the boundary condition
    q_flow = getConstReal(BC, 'q_flow', GotIt)
    IF (.NOT. GotIt) THEN
         print *, "q_flow not found in boundary condition"
         q_flow = 0.0        
    END IF 

    ! Add constant heat flux to boundary
    hf = hf + q_flow
```

Reply:
recommendation is added into module source file